be/src/vec/aggregate_functions/aggregate_function_collect.h

```diff
@@ -49,18 +49,16 @@ struct AggregateFunctionCollectSetData {
         data_set.insert(assert_cast<const ColVecType&>(column).get_data()[row_num]);
     }
 
-    void merge(const SelfType& rhs) { data_set.merge(rhs.data_set); }
-
-    void merge(const SelfType& rhs, bool has_limit) {
-        if (!has_limit) {
-            merge(rhs);
-            return;
-        }
-        for (auto& rhs_elem : rhs.data_set) {
-            if (size() >= max_size) {
-                return;
+    void merge(const SelfType& rhs) {
+        if constexpr (HasLimit::value) {
+            data_set.merge(rhs.data_set);
+        } else {
```

Comment:
the template condition seems to mistake in writing, have limit but merge all data

Reply:
OK, I'll check it out.